controllers/activemqartemis_controller.go

```diff
@@ -137,61 +137,56 @@ func (r *ActiveMQArtemisReconciler) Reconcile(ctx context.Context, request ctrl.
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			reqLogger.V(1).Info("ActiveMQArtemis Controller Reconcile encountered a IsNotFound, for request NamespacedName " + request.NamespacedName.String())
-			return ctrl.Result{}, nil
+			return result, nil
 		}
-		reqLogger.Error(err, "unable to retrieve the ActiveMQArtemis for request "+request.Namespace+"/"+request.Name)
-		return ctrl.Result{}, err
+		reqLogger.Error(err, "unable to retrieve the ActiveMQArtemis")
+		return result, err
 	}
 
 	namer := MakeNamers(customResource)
 	reconciler := ActiveMQArtemisReconcilerImpl{}
 
-	result := ctrl.Result{}
-	var valid = true
-
-	if valid, result = validate(customResource, r.Client, r.Scheme, *namer); valid {
+	var requeueRequest bool = false
+	var valid bool = false
+	if valid, requeueRequest = validate(customResource, r.Client, r.Scheme, *namer); valid {
 
 		err = reconciler.Process(customResource, *namer, r.Client, r.Scheme)
 
-		result = UpdateBrokerPropertiesStatus(customResource, r.Client, r.Scheme)
+		if ProcessBrokerStatus(customResource, r.Client, r.Scheme) {
+			requeueRequest = true
+		}
 	}
 
-	UpdateStatus(customResource, r.Client, request.NamespacedName, *namer, err)
+	ProcessStatus(customResource, r.Client, request.NamespacedName, *namer, err)
 
-	err = UpdateCRStatus(customResource, r.Client, request.NamespacedName)
-
-	if err != nil {
-		if apierrors.IsConflict(err) {
-			reqLogger.V(1).Info("unable to update ActiveMQArtemis status", "Request Namespace", request.Namespace, "Request Name", request.Name, "error", err)
-			err = nil // we don't want the controller event loop reporting this as an error
-		} else {
-			reqLogger.V(1).Error(err, "unable to update ActiveMQArtemis status", "Request Namespace", request.Namespace, "Request Name", request.Name)
-		}
-		return ctrl.Result{RequeueAfter: common.GetReconcileResyncPeriod()}, err
+	crStatusUpdateErr := UpdateCRStatus(customResource, r.Client, request.NamespacedName)
+	if crStatusUpdateErr != nil {
+		requeueRequest = true
 	}
 
-	if result.IsZero() {
+	if !requeueRequest {
 		reqLogger.Info("resource successfully reconciled")
 		if hasExtraMounts(customResource) {
-			reqLogger.V(1).Info("resource has extraMounts, requeuing for periodic sync")
-			result = ctrl.Result{RequeueAfter: common.GetReconcileResyncPeriod()}
+			reqLogger.V(1).Info("resource has extraMounts, requeuing")
+			requeueRequest = true
 		}
-	} else {
-		reqLogger.V(1).Info("requeue resource")
 	}
+
+	if requeueRequest {
+		reqLogger.V(1).Info("requeue reconcile")
+		result = ctrl.Result{RequeueAfter: common.GetReconcileResyncPeriod()}
+	}
+
 	return result, err
 }
 
-func validate(customResource *brokerv1beta1.ActiveMQArtemis, client rtclient.Client, scheme *runtime.Scheme, namer Namers) (bool, ctrl.Result) {
+func validate(customResource *brokerv1beta1.ActiveMQArtemis, client rtclient.Client, scheme *runtime.Scheme, namer Namers) (valid bool, retry bool) {
```

Comment:
Where are you using the named return arguments?

Reply:
I was only using one, the retry. Can remove the name from the first bool, will do that. thanks

Reply:
I did not use a naked return, but I think it helps to use the named argument in this case.